netscaler/netscaler_resource.go

```diff
@@ -151,12 +151,30 @@ func (c *NitroClient) createHTTPRequest(method string, urlstr string, buff *byte
 	return req, nil
 }
 
+func maskHeaders(headers http.Header) http.Header {
+	maskedHeaders := make(http.Header)
```

Comment:
`make(http.Header, len(headers)` would be slightly faster.

Reply:
also, no guarantee that headers is not nil?

Reply:
When you don't set any headers with
```go
req.Header.Set()
```
request headers default to an empty map:
```go
headers == map[]
```
The code of `maskHeaders()` needs no change to accommodate this use case since `range` will work correctly and result in a loop that is never run.
The return value will still be an empty map.